utils.c

```diff
@@ -18,10 +18,22 @@
 #include <dirent.h>
 #include <errno.h>
 #include <err.h>
+#include <unistd.h>
 
 #include "version.h"
 #include "utils.h"
 
+void restrict_fsaccess(const char **rootdir)
```

Comment:
Make this function available only for OpenBSD and have a function with only `UNUSED(rootdir);` for all other platforms.

Reply:
🤔 I know no \*nix w/o chroot(2) available. Why shouldn’t all of them profit from this?

Reply:
You are assuming that F3 only runs on *nix and that all *nixes provide all system calls. F3 runs on Windows for example.

If you do the test on all currently supported platforms and F3 runs, I can accept the `chroot()`.

Reply:
* Linux: ✅ / unistd.h / since SVr4 (Source: https://man7.org/linux/man-pages/man2/chroot.2.html)
* FreeBSD: ✅ / unistd.h / since Version 7 AT&T UNIX (Source: https://www.freebsd.org/cgi/man.cgi?query=chroot&sektion=2)
* MacOS: ✅ / unistd.h / since 4.2BSD (Source: https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/chroot.2.html)
* OpenBSD: ✅ / unistd.h / since Version 7 AT&T UNIX (Source: https://man.openbsd.org/chroot.2)
* Cygwin: Ok, let's say both of us were right. 😅 I with that all \*nix provide chroot(2) and you with that it's a good idea to test on Cygwin.

Reply:
Given your effort and the broad compatibility of `chroot()`, I'm open to code that makes `chroot()` available on all platforms but Cygwin.

It seems that making `restrict_fsaccess()` portable would be cleaner than making `chroot()` portable like the other portable functions in the code. But I'm open to ideas here.

Reply:
Err... chroot(2) _is_ portable. I've actually tested on Cygwin and the only problem was:

1. As you make /c/foo/bar your new root, your destination paths become /c/foo/bar/1.h2w -> /1.h2w.
2. full_fn_from_number() just assembles them a bit wrong: volume ("/") + separator ("/") + number + ".h2w"
3. The resulting //1.h2w isn’t a problem on \*nix _except_ Cygwin where // != /.

I've fixed that.

Reply:
Thank you for digging into the portability of `chroot()`.